### ISSUE TITLE
[linux] [MIDIOut] `.connect` with no argument connects to port 0

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#5771]
## Environment

* SuperCollider version: 3.12.x
* Operating system: Linux

## Steps to reproduce

```supercollider
m = MIDIOut(0).connect;
```

Now, in qjackctl, you'll see a connection to "Midi Through." :confused:

## Expected vs. actual behavior

As an opinion, I'd propose that `.connect` should connect to a specific device of the user's choice.

If the user didn't make a choice (omitted the argument), then there are three things we could do:

- Verbose failure: Throw an error.
- Silent failure: Do nothing and return without error.
- Arbitrary result: Connect to *something* without any indication to the user that this might be nonsense.

IMO the current behavior ("do nonsense") is the worst of these three, and the first option is probably the best.

Should we change this to an error?

**Comment by hitesh-kumain** [comment]
hi  @jamshark70 can you assign this to me ?

**Comment by dyfer** [comment]
Hello @hitesh-kumain 

You are welcome to simply submit a PR if you have a solution for this problem. Although, it seems that this particular issue needs further discussion to identify the proper solution.
We do use the "assign" feature, but not in every PR/issue, so feel free to comment/contribute as desired.

**Comment by jamshark70** [comment]
> Although, it seems that this particular issue needs further discussion to identify the proper solution.

I don't think it needs much discussion... a descriptive error message makes the most sense IMO.

@hitesh-kumain I agree with dyfer -- go ahead and put in a PR. Include "Fixes #5771" in the PR's body text to link back here and automatically close this issue when the PR is merged.

**Comment by hitesh-kumain** [comment]
Ya sure !
